src/main/java/net/clgd/ccemux/rendering/tror/TRoRRenderer.java

```diff
@@ -0,0 +1,217 @@
+								char current = payload.charAt(index++);
+
+								if (current == startChar) {
+									break;
+								} else if (current == '\\') {
+									current = payload.charAt(index++);
+									if (current == 'a') builder.append((char) 0x7);
+									else if (current == 'b') builder.append('\b');
+									else if (current == 'f') builder.append('\f');
+									else if (current == 'n') builder.append('\n');
+									else if (current == 'r') builder.append('\r');
+									else if (current == 't') builder.append('\t');
+									else if (current == 'v') builder.append((char) 0xB);
+									else if (current == '\\') builder.append('\\');
+									else if (current == '\'') builder.append('\'');
+									else if (current == '\"') builder.append('\"');
+
+									// TODO: Support for numeric escape codes.
+								} else {
+									builder.append(current);
+								}
+							}
+
+							args.add(builder.toString());
+							if (index >= payload.length() - 1 || payload.charAt(index) == ',') index++;
+						} else {
+							int next = payload.indexOf(',', index);
+							if (next < 0) next = payload.length();
+							String arg = payload.substring(index, next);
+							switch (arg) {
+								case "nil":
+									args.add(null);
+									break;
+								case "true":
+									args.add(true);
+									break;
+								case "false":
+									args.add(false);
+									break;
+								default:
+									try {
+										args.add(Double.parseDouble(arg));
+									} catch (NumberFormatException e) {
+										args.add(null);
+									}
+									break;
+							}
+
+							index = next + 1;
+						}
+					}
+
+					computer.queueEvent(event, args.toArray());
+				}
+			}
+		}
+	}
+
+	private void sendLine(String mode, String line) {
+		try {
+			output.write(mode);
+			output.write(':');
+			output.write(Integer.toString(computer.getID()));
+			output.write(';');
+			output.write(line);
+			output.write('\n');
+			output.flush();
+		} catch (IOException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	@Override
+	public void write(String text) {
+		sendLine("TW", text.replace('\r', ' ').replace('\n', ' '));
+	}
+
+	@Override
+	public void setCursorPos(int x, int y) {
+		sendLine("TC", x + "," + y);
+	}
+
+	@Override
+	public void clear() {
+		sendLine("TE", "");
+	}
+
+	@Override
+	public void clearLine() {
+		sendLine("TL", "");
+	}
+
+	@Override
+	public void scroll(int yDiff) {
+		sendLine("TS", "");
+	}
+
+	@Override
+	public void setCursorBlink(boolean blink) {
+		sendLine("TB", blink ? "true" : "faalse");
+	}
+
+	@Override
+	public void setTextColour(int colour) {
+		sendLine("TF", COLOURS[colour]);
+	}
+
+	@Override
+	public void setBackgroundColour(int colour) {
+		sendLine("TK", COLOURS[colour]);
+	}
+
+	@Override
+	public void setColour(int index, double r, double g, double b) {
+		sendLine("TM", String.format("%s,%.4f,%.4f,%.4f", COLOURS[index], r, g, b));
+	}
+
+	@Override
+	public void resize(int width, int height) {
```

Comment:
There's a bit of a conflict here, `Renderer.resize` is intended to be used to tell the renderer to resize itself to a certain size (although it's unused and will be removed soon, so probably not an issue)